ckanext/saml2/plugin.py

```diff
@@ -653,3 +660,18 @@ def slo(self):
     def staff_login(self):
         """Default login page for staff members."""
         return self.login()
+
+    def edit(self, id=None, data=None, errors=None, error_summary=None):
+        c = p.toolkit.c
+        c.allow_user_changes = config.get('ckan.saml2.allow_user_changes', False)
+        return super(Saml2Controller, self).edit(id, data, errors, error_summary)
+
+    def _save_edit(self, id, context):
+        data_dict = logic.clean_dict(unflatten(
+            logic.tuplize_dict(logic.parse_params(request.params))))
```

Comment:
Are you sure, that you have no access to `request.POST` ? This construction is quite cumbersome